timefold-solver-python-core/src/main/python/score/_group_by.py

```diff
@@ -993,6 +996,34 @@ def to_sorted_map(key_mapper, value_mapper, merge_function_or_set_creator=None):
         else:
             raise ValueError
 
+    @overload
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A], Balanced], load_function: Callable[[A], int] = None,
+                     initial_load_function: Callable[[A], int] = None) -> \
+            'UniConstraintCollector[A, Any, LoadBalance[Balanced]]':
+        ...
+
+    @overload
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B], Balanced], load_function: Callable[[A, B], int] = None,
+                     initial_load_function: Callable[[A, B], int] = None) -> \
+            'BiConstraintCollector[A, B, Any, LoadBalance[Balanced]]':
+        ...
+
+    @overload
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C], Balanced], load_function: Callable[[A, B, C], int] = None,
+                     initial_load_function: Callable[[A, B, C], int] = None) -> \
+            'TriConstraintCollector[A, B, C, Any, LoadBalance[Balanced]]':
+        ...
+
+    @overload
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C, D], Balanced],
+                     load_function: Callable[[A, B, C, D], int] = None,
+                     initial_load_function: Callable[[A, B, C, D], int] = None) -> \
+            'QuadConstraintCollector[A, B, C, D, Any, LoadBalance[Balanced]]':
+        ...
 
```

Comment:
Pretty sure this is wrong/incomplete. Yet no tests are failing.

- How do I make this correct?
- Why is no test failing?

Reply:
- To make it correct, add a "collector holder" here:
  https://github.com/TimefoldAI/timefold-solver-python/blob/5aa2cbd6fbeddd296776ca59ba8fefcd464944d7/timefold-solver-python-core/src/main/python/score/_group_by.py#L57-L63
  like this:
  ```python
   @dataclasses.dataclass 
   class LoadBalanceCollector: 
       collector_creator: Callable 
       balanced_item_function: Callable 
       load_function: Callable | None
       initial_load_function: Callable  | None
  ```
  and create a converter to translate the holder here:
  https://github.com/TimefoldAI/timefold-solver-python/blob/5aa2cbd6fbeddd296776ca59ba8fefcd464944d7/timefold-solver-python-core/src/main/python/score/_group_by.py#L64-L91
  like this:
  ```python
    elif isinstance(collector_info, LoadBalanceCollector):
        balanced_item_function = function_cast(collector_info.mapping_function,  *type_arguments)
        if collector_info.load_function is None:
           return collector_info.collector_creator(balanced_item_function)
        load_function = function_cast(collector_info.load_function,  *type_arguments)
        if collector_info.initial_load_function is None:
           return collector_info.collector_creator(balanced_item_function, load_function)
        initial_load_function =  function_cast(collector_info.initial_load_function,  *type_arguments)
        return collector_info.collector_creator(balanced_item_function, load_function, initial_load_function)
  ```
- No test is failing because there no test for load balance, because automatically testing equivalent functionality is not feasible (so only an existence test is done).